src/http/keys/keys.controller.ts

```diff
@@ -34,9 +39,29 @@ export class KeysController {
     description: 'List of all keys',
     type: KeyListResponse,
   })
-  @ApiOperation({ summary: 'Get list of all keys' })
-  get(@Query() filters: KeyQuery) {
-    return this.keysService.get(filters);
+  @ApiOperation({ summary: 'Get list of all keys in stream' })
+  async get(@Query() filters: KeyQuery, @Res() reply: FastifyReply) {
+    // Because the real execution of generators occurs in the controller's method, that's why we moved the transaction here
+    await this.entityManager.transactional(
+      async () => {
+        const { keysGenerators, meta } = await this.keysService.get(filters);
+
+        const jsonStream = JSONStream.stringify('{ "meta": ' + JSON.stringify(meta) + ', "data": [', ',', ']}');
+        reply.type('application/json').send(jsonStream);
+        // TODO: is it necessary to check the error? or 'finally' is ok?
+        try {
+          for (const keysGenerator of keysGenerators) {
+            for await (const key of keysGenerator) {
+              const keyReponse = new Key(key);
+              jsonStream.write(keyReponse);
```

Comment:
https://linear.app/lidofi/issue/VAL-413/right-way-to-catch-exceptions-during-streaming